source/browseMode.py

```diff
@@ -196,6 +196,44 @@ def isAfterSelection(self):
 		caret=self.document.makeTextInfo(textInfos.POSITION_CARET)
 		return self.textInfo.compareEndPoints(caret, "startToStart") > 0
 
+	def _getLabelForProperties(self, labelPropertyGetter):
+		"""
+		Fetches required properties for this L{TextInfoQuickNavItem} and constructs a label to be shown in an elements list.
+		@Param labelPropertyGetter: A callable taking 1 argument.
+			For example, if L{itemType} is landmark, the callable must return the landmark type when the landmark argument is provided to it.
+			Alternative attributes or dictionary keys might be name or value.
+			An expected callable might be the __getattribute__ method on an L{NVDAObject}, or the get method on a L{Dict}.
+			Note that L{Dict.get} doesn't raise an exception, it returns None instead.
+		"""
+		value = self.textInfo.text.strip()
+		if self.itemType is "heading":
+			return value
+		labelParts = None
+		name = labelPropertyGetter("name")
+		if self.itemType is "landmark":
+			landmark = aria.landmarkRoles.get(labelPropertyGetter("landmark"))
+			labelParts = (name, landmark)
+		else: 
+			role = labelPropertyGetter("role")
+			roleText = controlTypes.roleLabels[role]
+			# Translators: Reported label in the elements list for an element which which has no name and value
+			unlabeled = _("Unlabeled")
+			realStates = labelPropertyGetter("states")
+			positiveStates = " ".join(controlTypes.stateLabels[st] for st in controlTypes.processPositiveStates(role, realStates, controlTypes.REASON_FOCUS, realStates))
+			negativeStates = " ".join(controlTypes.negativeStateLabels[st] for st in controlTypes.processNegativeStates(role, realStates, controlTypes.REASON_FOCUS, realStates))
+			if self.itemType is "formField":
+				if role in (controlTypes.ROLE_BUTTON,controlTypes.ROLE_DROPDOWNBUTTON,controlTypes.ROLE_TOGGLEBUTTON,controlTypes.ROLE_SPLITBUTTON,controlTypes.ROLE_MENUBUTTON,controlTypes.ROLE_DROPDOWNBUTTONGRID,controlTypes.ROLE_SPINBUTTON,controlTypes.ROLE_TREEVIEWBUTTON):
+					labelParts = (value or name or unlabeled, roleText, positiveStates, negativeStates)

```

Comment:
It'd be good to have a comment with example content that might be produced by each of these conditions:

> `# For example: "Mute toggle button pressed"`